airbyte-server/src/main/java/io/airbyte/server/handlers/OAuthHandler.java

```diff
@@ -207,4 +254,36 @@ private Map<String, Object> generateDestinationMetadata(final UUID destinationDe
     return TrackingMetadata.generateDestinationDefinitionMetadata(destinationDefinition);
   }
 
+  @VisibleForTesting
+  List<String> buildJsonPathFromOAuthFlowInitParameters(List<List<String>> oAuthFlowInitParameters) {
+    return oAuthFlowInitParameters.stream()
+        .map(path -> "$." + String.join(".", path))
+        .toList();
+  }
+
+  @VisibleForTesting
+  JsonNode getOauthFromDBIfNeeded(JsonNode oAuthInputConfigurationFromDB, JsonNode oAuthInputConfigurationFromInput) {
+    Map<String, String> result = new HashMap<>();
+
+    Jsons.deserializeToStringMap(oAuthInputConfigurationFromInput)
+        .forEach((k, v) -> {
+          if (AirbyteSecretConstants.SECRETS_MASK.equals(v)) {
+            result.put(k, oAuthInputConfigurationFromDB.get(k).textValue());
```

Comment:
Are we guaranteed that `oAuthInputConfigurationFromDB` will always have an entry for each key here? If not, this could result in an NPE, so we should probably handle that case explicitly

Reply:
I don't know. I will add a check in case if it is null.

Reply:
Done